swarm-cluster-using-docker-extension/azuredeploy.json

```diff
@@ -0,0 +1,396 @@
+				"description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+			}
+		},
+		"adminUsername": {
+			"type": "string",
+			"metadata": {
+				"description": "User name for the Virtual Machine."
+			}
+		},
+		"adminPassword": {
+			"type": "securestring",
+			"metadata": {
+				"description": "Password for the Virtual Machine."
+			}
+		},
+		"dnsNameForPublicIP": {
+			"type": "string",
+			"metadata": {
+				"description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+			}
+		},
+		"ubuntuOSVersion": {
+			"type": "string",
+			"defaultValue": "14.10",
+			"allowedValues": [
+				"12.04.2-LTS",
+				"12.04.3-LTS",
+				"12.04.4-LTS",
+				"12.04.5-LTS",
+				"12.10",
+				"14.04.2-LTS",
+				"14.10",
+				"15.04"
+			],
+			"metadata": {
+				"description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version. Allowed values: 12.04.2-LTS, 12.04.3-LTS, 12.04.4-LTS, 12.04.5-LTS, 12.10, 14.04.2-LTS, 14.10, 15.04."
+			}
+		},
+		"slaveCount": {
+			"type": "int",
+			"defaultValue": 1,
+			"metadata": {
+				"description": "Number of slave nodes"
+			}
+		},
+		"swarmClusterId": {
+			"type": "string",
+			"metadata": {
+				"description": "Swarm cluster id"
+			}
+		}
+    },
+    "variables": {
+    		"location": "West US",
+    		"extensionName": "DockerExtension",
+    		"storageAccountType": "Standard_LRS",
+    		"publicIPAddressName": "swarmPublicIP",
+    		"publicIPAddressType": "Dynamic",
+    		"nicName": "swarmVMNic",
+    		"virtualNetworkName": "SwarmVNET",
+		"subnetName": "subnet",
+		"addressPrefix": "10.0.0.0/16",
+		"subnetPrefix": "10.0.0.0/24",
+    		"vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+		"subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+		"vmName": "swarmVM",
+		"vmSize": "Standard_D1",
+		"imagePublisher": "Canonical",
+		"imageOffer": "UbuntuServer",
+		"OSDiskName": "osdiskforlinuxsimple",
+		"vmStorageAccountContainerName": "vhds",
+            	"nsgName" : "swarmNSG",
+            	"nsgID": "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]"
+    },
+    "resources": [
+    	{
+		"type": "Microsoft.Storage/storageAccounts",
+		"name": "[parameters('newStorageAccountName')]",
+		"apiVersion": "2015-05-01-preview",
+		"location": "[variables('location')]",
+		"properties": {
+			"accountType": "[variables('storageAccountType')]"
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
+		"type": "Microsoft.Network/publicIPAddresses",
+		"name": "[concat(variables('publicIPAddressName'), 'master')]",
+		"location": "[variables('location')]",
+		"properties": {
+			"publicIPAllocationMethod": "[variables('publicIPAddressType')]",
+			"dnsSettings": {
+				"domainNameLabel": "[concat(parameters('dnsNameForPublicIP'), 'master')]"
+			}
+		}
+	},
+	{
+		"apiVersion": "2015-05-01-preview",
```

Comment:
Do we even need public IP addresses for the individual slaves? I guess for debugging.